cla-backend/cla/controllers/github.py

```diff
@@ -759,21 +759,68 @@ def validate_organization(body):
             return {"status": "error"}
 
 
-def webhook_secret_validation(webhook_signature, data):
+def webhook_secret_validation(webhook_signature: str, data: bytes) -> bool:
+    """
+    webhook_secret_validation checks if webhook_signature is same as incoming data's
+    :param webhook_signature:
+    :param data:
+    :return:
+    """
+    cla.log.debug(f"webhook_secret_validation for signature {webhook_signature} and env : {cla.config.GITHUB_APP_WEBHOOK_SECRET}")
+    if cla.config.GITHUB_APP_WEBHOOK_SECRET == "":
+        raise RuntimeError("GITHUB_APP_WEBHOOK_SECRET is empty")
+
     if not webhook_signature:
         return False
 
     sha_name, signature = webhook_signature.split('=')
-
     if not sha_name == 'sha1':
         return False
 
-    mac = hmac.new(os.environ.get('GH_APP_WEBHOOK_SECRET', '').encode('utf-8'), msg=data, digestmod='sha1')
-    pprint(str(mac.hexdigest()))
-    pprint(str(signature))
-    pprint(data)
+    mac = hmac.new(cla.config.GITHUB_APP_WEBHOOK_SECRET.encode('utf-8'), msg=data, digestmod='sha1')
+    hex_digest = mac.hexdigest()
+    return True if hmac.compare_digest(hex_digest, signature.strip()) else False
+
+
+def webhook_secret_failed_email_content(event_type: str, req_body: dict, maintainers: List[str]):
+    """Helper function to update maintainers about failed webhook secrets"""
+    if not maintainers:
+        cla.log.warning("webhook_secret_failed_email - maintainers list is empty can't send the email.")
+        raise RuntimeError("no maintainers set")
+
+    user_login = req_body.get('sender', {}).get('login', None)
+    repository_id = req_body.get('repository', {}).get('id', None)
+    repository_name = req_body.get('repository', {}).get('full_name', None)
+    installation_id = req_body.get('installation', {}).get('id', None)
+    msg = f"""webhook secret validation failed :
+    event type: {event_type},
+    user login: {user_login},
+    repository_id: {repository_id}, repository_name : {repository_name},
+    installation_id: {installation_id}"""
 
-    return True if hmac.compare_digest(mac.hexdigest(), signature) else False
+    body = f"""
+    <p>Hello EasyCLA Maintainer,</p>
+    <p>This is a notification email from EasyCLA regarding failure of webhook secret validation.</p>
+    <p>{msg}</p>"""
```

Comment:
Very good, but we want to add the standard help doc/links and the sign-off line to the end of the system generated emails.

Reply:
got it 👍